.github/workflows/weekly_tests.yaml

```diff
@@ -0,0 +1,23 @@
+name: Weekly Tests
+
+on:
+  schedule:
+    - cron: "0 0 * * SAT"
```

Comment:
From experience, it would be better choosing a more arbitrary time to avoid resource saturation (e.g. other system level processes or tests from other teams scheduled at the same time)

Reply:
Agree. We had discussed this topic but we don't have good solution now. Randomization is not suppprted in Github scheduled workflow.

Reply:
I'm currently looking at ways to resolve this. Will update with possible solutions in a bit.

Reply:
Discussions for possible solutions are present on this (internal) [Mattermost thread](https://chat.canonical.com/canonical/pl/ws3n43y3d3bjz8iweoxeypdpfh).

[Random sleep workflow](https://pastebin.ubuntu.com/p/nK5CbgNxTn/) idea.